doctr/models/recognition/vip/pytorch.py

```diff
@@ -0,0 +1,285 @@
+from .base import _VIPTR, _VIPTRPostProcessor
+
+__all__ = ["VIPTRPostProcessor", "VIPTR", "viptr_base", "viptr_tiny"]
+
+
+class VIPTRPostProcessor(_VIPTRPostProcessor):
+    """Postprocess raw prediction of the model (logits) to a list of words using CTC decoding."""
+
+    @staticmethod
+    def ctc_best_path(
+        logits: torch.Tensor,
+        vocab: str,
+        blank: int = 0,
+    ) -> List[Tuple[str, float]]:
+        """
+        Implements best path decoding as shown by Graves (Dissertation, p63).
+
+        Args:
+            logits: model output, shape: N x T x C
+            vocab: vocabulary string
+            blank: index of blank label
+
+        Returns:
+            List of (word, confidence) for each sample
+        """
+        # Compute a "confidence" for each sequence
+        probs = F.softmax(logits, dim=-1).max(dim=-1).values.min(dim=1).values
+        # Collapse best path (itertools.groupby), map indices->chars, join to string
+        preds_indices = torch.argmax(logits, dim=-1)
+        print(preds_indices.max())
+        words = [decode_sequence([k for k, _ in groupby(seq.tolist()) if k != blank], vocab) for seq in preds_indices]
+
+        return list(zip(words, probs.tolist()))
+
+    def __call__(self, logits: torch.Tensor) -> List[Tuple[str, float]]:
+        """
+        Decodes CTC logits into strings plus confidence.
+
+        Args:
+            logits: raw output of shape (N, T, C)
+
+        Returns:
+            List of (word, confidence)
+        """
+        return self.ctc_best_path(logits=logits, vocab=self.vocab, blank=0)
+
+
+class VIPTR(_VIPTR, nn.Module):
+    """
+    Main VIPTR model class for text recognition.
+    """
+
+    def __init__(
+        self,
+        feature_extractor: nn.Module,
+        vocab: str,
+        embedding_units: int = 384,
+        input_shape: Tuple[int, int, int] = (3, 32, 128),
+        output_channel: int = 192,
+        cfg: Optional[Dict[str, Any]] = None,
+        exportable: bool = False,
+        max_length: int = 32,
+    ):
+        """
+        Args:
+            feature_extractor: backbone feature extractor
+            vocab: string containing all supported characters
+            embedding_units: dimension of embeddings
+            input_shape: (C, H, W)
+            output_channel: optional final output channels
+            cfg: optional config
+            exportable: whether to mark the model export-friendly
+            max_length: max sequence length
+        """
+        super().__init__()
+        self.vocab = vocab
+        self.exportable = exportable
+        self.cfg = cfg
+        self.max_length = max_length
+        self.vocab_size = len(vocab)
+
+        self.feat_extractor = feature_extractor
+        self.postprocessor = VIPTRPostProcessor(vocab=self.vocab)
+        self.head = nn.Linear(embedding_units, len(self.vocab) + 1)  # +1 for PAD
+        self.ctc_loss = nn.CTCLoss(zero_infinity=True)
```

Comment:
as mentioned better to go with the functional implementation and move to compute_loss then we can keep it as staticmethod